integration/container_stats_test.go

```diff
@@ -0,0 +1,317 @@
+		}()
+		require.NoError(t, runtimeService.StartContainer(cn))
+		defer func() {
+			assert.NoError(t, runtimeService.StopContainer(cn, 10))
+		}()
+	}
+
+	t.Logf("Fetch container stats for each containerd with Filter")
+	var stats []*runtime.ContainerStats
+	require.NoError(t, Eventually(func() (bool, error) {
+		stats, err = runtimeService.ListContainerStats(
+			&runtime.ContainerStatsFilter{PodSandboxId: sb})
+		if err != nil {
+			return false, err
+		}
+		if len(stats) != 3 {
+			return false, fmt.Errorf("unexpected stats length")
+		}
+		if stats[0].GetWritableLayer().GetUsedBytes().GetValue() != 0 &&
+			stats[0].GetWritableLayer().GetInodesUsed().GetValue() != 0 {
+			return true, nil
+		}
+		return false, nil
+	}, time.Second, 30*time.Second))
+	t.Logf("Verify container stats for sandbox %q", sb)
+	for _, s := range stats {
+		testStats(t, s)
+	}
+}
+
+// Test to verify filtering given a specific container ID and
+// sandbox ID
+func TestContainerListStatsWithIdSandboxIdFilter(t *testing.T) {
+	t.Logf("Create a pod config and run sandbox container")
+	sbConfig := PodSandboxConfig("running-pod", "statsls")
+	sb, err := runtimeService.RunPodSandbox(sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sb))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sb))
+	}()
+	t.Logf("Create container config and run containers in a pod")
+	containers := []string{}
+	opt := func() ContainerOpts {
+		return func(cf *runtime.ContainerConfig) {
+			cf.Labels = map[string]string{"key": "value"}
+			cf.Annotations = map[string]string{"a.b.c": "test"}
+		}
+	}()
+	for i := 0; i < 3; i++ {
+		cName := fmt.Sprintf("container%d", i)
+		containerConfig := ContainerConfig(cName, defaultTestImage, opt)
+		cn, err := runtimeService.CreateContainer(sb, containerConfig, sbConfig)
+		containers = append(containers, cn)
+		require.NoError(t, err)
+		defer func() {
+			assert.NoError(t, runtimeService.RemoveContainer(cn))
+		}()
+		require.NoError(t, runtimeService.StartContainer(cn))
+		defer func() {
+			assert.NoError(t, runtimeService.StopContainer(cn, 10))
+		}()
+	}
+	t.Logf("Fetch container stats for sandbox ID and container ID filter")
+	var stats []*runtime.ContainerStats
+	require.NoError(t, Eventually(func() (bool, error) {
+		stats, err = runtimeService.ListContainerStats(
+			&runtime.ContainerStatsFilter{Id: containers[0], PodSandboxId: sb})
+		if err != nil {
+			return false, err
+		}
+		if len(stats) != 1 {
+			return false, fmt.Errorf("Unexpected stats length")
+		}
+		if stats[0].GetWritableLayer().GetUsedBytes().GetValue() != 0 &&
+			stats[0].GetWritableLayer().GetInodesUsed().GetValue() != 0 {
+			return true, nil
+		}
+		return false, nil
+	}, time.Second, 30*time.Second))
+	t.Logf("Verify container stats for sandbox %q and container %q filter", sb, containers[0])
+	for _, s := range stats {
+		testStats(t, s)
+	}
+}
+
+func testStats(t *testing.T, s *runtime.ContainerStats) {
```

Comment:
Pass in container config for verification? Even for `ListContainerStats`, we could find corresponding container config inside the list and pass in.